src/v/kafka/requests/kafka_batch_adapter.h

```diff
@@ -40,18 +40,22 @@ constexpr size_t kafka_header_size = sizeof(int64_t) + // base offset
  * Usage:
  *
  *   kafka_batch_adapter kba;
- *   kba.adapt(std::move(batch));
+ *   while(batch = kba.adapt(std::move(batch)); !batch.empty()) {
```

Comment:
confused about returning a batch from kba.adapt, but then also doing something with kba.batch.

Reply:
i see now that the returned `batch` is an iobuf and `kba.batch` is a batch. maybe change `batch` -> `remaininder`?